<?xml version='1.0'?>
<launch>
	<arg name="image_topic" default="/webcam/image_raw"/>
	<arg name="camera_info_topic" default="/webcam/camera_info"/>
	<arg name="compressed" default="False"/>

	 <!--  -->
	<node pkg="ml_detector" type="ml_detector_node" name="ml_detector" clear_params="true" output="screen">
		<!-- System Parameters -->
		<param name="system/publish_overlay" value="true" type="boolean"/>
		<param name="system/send_detailed_tag_info" value="true" type="boolean"/>
		<param name="system/show_rejected" value="false" type="boolean"/>
		<param name="system/refine_strategy" value="true" type="boolean"/>
		<param name="system/camera_is_rectified" value="false" type="boolean"/>

		<!-- Detection Parameters /-->
		<rosparam command="load" file="$(find ml_detector)/launch/detect_config.yaml"/>
		<rosparam command="load" file="$(find ml_detector)/launch/board_config.yaml"/>

		<!-- Marker Parameters -->
		<remap from="~detected_markers" to="/ml_landmarks/detected_markers"/>

		<!-- Image Parameters -->
		<remap from="~camera_info" to="$(arg camera_info_topic)"/>
		<remap from="~image_raw" to="$(arg image_topic)"/>

		<param if="$(arg compressed)" name="image_transport" value="compressed"/>
	</node>
</launch>
